runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCachePartition.java

```diff
@@ -496,10 +498,12 @@ public void writeEntryStart(
                 KafkaCachePaddedKeyFW paddedKey =
                     logFile.readBytes(position, paddedKeyRO::wrap);
                 final int paddedKeySize = paddedKey.sizeof();
-                KafkaCachePaddedKeyFW.Builder paddedKeyBuilder = paddedKeyRW;
-                final int keySize = paddedKeyBuilder.key(k -> k.length(length).value(buffer, index, length)).sizeof();
-                paddedKeyBuilder.padding(logFile.buffer(), 0, paddedKeySize - keySize - SIZE_OF_INT);
-                KafkaCachePaddedKeyFW newPaddedKey = paddedKeyBuilder.build();
+                paddedKeyRW.rewrap();
+                KafkaKeyFW keyFW = keyRW.rewrap().length(length).value(buffer, index, length).build();
+                paddedKeyRW.key(keyFW);
+                final int keyAndMetadataSize = keyFW.sizeof() + KafkaCachePaddedKeyFW.FIELD_OFFSET_PADDING;
+                paddedKeyRW.padding(logFile.buffer(), 0, paddedKeySize - keyAndMetadataSize);
+                KafkaCachePaddedKeyFW newPaddedKey = paddedKeyRW.build();
```

Comment:
IIUC, the intent of this change is to fix an issue with `paddedKeyRW` reuse, which can be simplified to the following:
```suggestion
                KafkaCachePaddedKeyFW.Builder paddedKeyBuilder = paddedKeyRW.rewrap();
                final int keySize = paddedKeyBuilder.key(k -> k.length(length).value(buffer, index, length)).sizeof();
                paddedKeyBuilder.padding(logFile.buffer(), 0, paddedKeySize - keySize - SIZE_OF_INT);
                KafkaCachePaddedKeyFW newPaddedKey = paddedKeyBuilder.build();
                logFile.writeBytes(position, newPaddedKey.buffer(), newPaddedKey.offset(), newPaddedKey.sizeof());
```
Agree?